dts/bindings/usb-c/vif.yaml

```diff
@@ -0,0 +1,113 @@
+# Copyright 2022 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    A VIF node is a root element that describes the Qualifying Product in VIF
+    specification. It contains introductory material followed by an optional
+    product-level field element, zero or more Components, and zero or more
+    Optional Content elements.
+    This is based on VIF XML file schema in:
+      https://compliance.usb.org/cv/VendorInfoFile/Schemas/Current/VendorInfoFile.xsd
+
+compatible: "vif"
```

Comment:
There are some high level specifications required for VIF, specific to product but applicable to all ports. I see USB-C DT nodes are per port, hence added this node, which is referring to the ports (i.e, usb-c connector nodes).

Reply:
@madhurimaparuchuri : 

Without commenting this is appropriate to include in devicetree (I am no longer maintaining the devicetree bindings so that I can focus on other tasks), I will say that choosing "vif" for the compatible is pretty clear namespace pollution. The namespace of compatible property values is shared across all hardware in the world, so a three-letter acronym had better be for something incredibly general purpose, which this definitely is not.

Please choose a better, more descriptive name, and make sure to expand the acronym (I guess it's "vendor info file") in the description field in the binding.

In general, please remember that you are the VIF domain expert here, but this binding lives in a larger world where there's lots of other hardware and people who will read your documentation without your domain knowledge.

Reply:
> @galak to review and decide on this

This is the decision maker on whether to put this in DT, since he is overall maintainer for the bindings.

I now only co-maintain the DT infrastructure and API.

Reply:
> I now only co-maintain the DT infrastructure and API.

I know, but given your experience your feedback serves as very valid input (at least better input than my feedback) on whether this approach looks appropriate for bindings.

Reply:
I'm inclined to agree that the devicetree should not directly represent the VIF fields for a few reasons:
1. The USB Implementers' Forum revs the VIF format somewhat regularly to keep up with new USB features. Sometimes, existing fields change name or format without changing the underlying meaning. Keeping up with these changes in the devicetree would involve changing a public-facing (sort of) API in the devicetree bindings.
2. Many of the fields will be unknowable to the Zephyr build system, and which ones are knowable depends on the hardware. To produce a valid VIF for, say, compliance testing, a developer would need to add in additional fields from some other source. If I were adding those fields, I'd probably want to use a format other than devicetree.
3. The bindings need to describe the devicetree properties, so they need to be verbose like the spec or refer to the spec or summarize the spec at the risk of vagueness or inaccuracy. I'd rather avoid the issue.
4. The VIF contains some parallel structure to the existing devicetree, e.g. different pieces for different ports. I'd prefer not to replicate that and risk them being inconsistent.

I'd prefer that the nodes for the port and individual port components contain properties for VIF-relevant information that is determined by the hardware, and the VIF generator would parse those properties (probably among other data sources) while constructing the VIF.

Reply:
I avoided adding elements to device tree. I am reading properties that are already there in device tree and generating the VIF, not adding any properties or nodes in this revision. The information that is there in VIF node in earlier revisions is being read form an XML file (manually added) in project folder if it's present (overriding the output VIF file).

Reply:
@tejlmand 
Could you review the new revision and let me know if any changes are required in this PR